### PR TITLE
Cache `Bundle` from `BundleToken` for performance

### PR DESCRIPTION
**Pull request by shuoli84**
* [x] I've started my branch from the `develop` branch (gitflow)
  * [x] And I've selected `develop` as the "base" branch for this Pull Request I'm about to create
* [x] I've added an entry in the `CHANGELOG.md` file to explain my changes and credit myself  
      _(or added `#trivial` to the PR title if I think it doesn't warrant a mention in the CHANGELOG)_
  * [x] Add the entry in the appropriate section (Bug Fix / New Feature / Breaking Change / Internal Change)
  * [x] Add a period and 2 spaces at the end of your short entry description
  * [x] Add links to your GitHub profile to credit yourself and to the related PR and issue after your description

---

### CODE REVIEW

**Comment by shuoli84** [comment]
fixes #623

**Comment by shuoli84** [comment]
fixes #623

**Comment by shuoli84** [comment]
Not sure why, the commit of modify contribution failed. @djbe any idea?

File: templates/strings/structured-swift4.stencil [hunk 1]
```diff
@@ -82,7 +82,13 @@ extension {{enumName}} {
   }
```
Comment (djbe) [inline templates/strings/structured-swift4.stencil:82]: You're not using the cached bundle in this template.

File: templates/strings/structured-swift3.stencil [hunk 2]
```diff
@@ -82,7 +82,13 @@ extension {{enumName}} {
   }
```
Comment (djbe) [inline templates/strings/structured-swift3.stencil:82]: You're not using the cached bundle in this template.

File: templates/strings/flat-swift4.stencil [hunk 3]
```diff
@@ -77,7 +77,13 @@ extension {{enumName}} {
   }
```
Comment (djbe) [inline templates/strings/flat-swift4.stencil:77]: You're not using the cached bundle in this template.

File: templates/strings/flat-swift3.stencil [hunk 4]
```diff
@@ -77,7 +77,13 @@ extension {{enumName}} {
   }
```
Comment (djbe) [inline templates/strings/flat-swift3.stencil:77]: You're not using the cached bundle in this template.

File: templates/ib/scenes-swift4.stencil [hunk 5]
```diff
@@ -90,7 +90,13 @@ import {{module}}
   }
```
Comment (djbe) [inline templates/ib/scenes-swift4.stencil:90]: You're not using the cached bundle in this template.

File: templates/ib/scenes-swift3.stencil [hunk 6]
```diff
@@ -88,7 +88,13 @@ import {{module}}
   }
```
Comment (djbe) [inline templates/ib/scenes-swift3.stencil:88]: You're not using the cached bundle in this template.

File: templates/fonts/swift3.stencil [hunk 7]
```diff
@@ -79,7 +79,13 @@
   }
 }
 
-private final class BundleToken {}
+// swiftlint:disable convenience_type
+private final class BundleToken {
+  static var bundle: Bundle = {
```
Comment (djbe) [inline templates/fonts/swift3.stencil]: This should be a `static let` I think. (applies to the other files as well)

**Review by djbe — CHANGES_REQUESTED** [review]
I'll take a look at why danger is failing for some reason.

File: templates/strings/structured-swift4.stencil (hunk 1 above)
Comment (shuoli84) [reply templates/strings/structured-swift4.stencil:82]: Yea.. I missed a lot places... Use checked with rg and should be good. Pls review again.

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by shuoli84** [comment]
@djbe pls check again?

**Review by djbe — APPROVED** [review]
Looks good 👍 

The `convenience_type` SwiftLint issue here is a bit annoying, but there's no good way to avoid it. Subclassing `NSObject` would avoid the warning, but seems a bit iffy to me.

**Comment by djbe** [comment]
Note that even though we've modified the strings template to be dynamic if needed (in #573), the provided bundle can be static, as it's the bundle of the app/framework, not the language bundle inside of it.

**Comment by shuoli84** [comment]
@djbe Not sure I fully understands the issue. If the framework itself is a static bundle, then it would merged into app bundle? Is it the same problem as https://stackoverflow.com/questions/2335664/localization-of-strings-in-static-lib? If that the case, maybe the bundle should be able to pass in, but with a reasonable default value.

**Comment by shuoli84** [comment]
Maybe create a test case to cover the case?

**Comment by djbe** [comment]
Nah, I think you misunderstood my comment. I was just explaining that, for users of the `localizeFunction` parameter (#573), this doesn't break things for them.

What you mentioned (a separate or specific bundle) is not something we support yet. For those use cases we recommend using a custom template, or they can always propose adding such functionality in a later PR.

**Comment by shuoli84** [comment]
Got it.

**Comment by shuoli84** [comment]
@djbe If there is no open issue need to address, then pls merge the pr? :P

**Pull request merged**